pennylane/pauli/conversion.py

```diff
@@ -346,6 +346,15 @@ def pauli_sentence(op):
     return _pauli_sentence(op)
 
 
+def is_pauli_sentence(op):
+    """Returns True of the operator is a PauliSentence and False otherwise."""
+    if op._pauli_rep is not None:  # pylint: disable=protected-access
+        return True
+    if not hasattr(op, "ops"):
```

Comment:
I was mimicking that [condition](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/pauli/conversion.py#L400) which seemed more general. If `Hamiltonian` objects are guaranteed to be `PauliSentence`s we could use your condition. Worse case scenario here is we'll catch it later when calling `_pauli_sentence`. On the other hand, it could handle stuff that is not intended to be a `PauliSentence`, but is.

Reply:
not sure what you mean. we only hit that line of code if `op` is a Hamiltonian, and then we'll know that it has the `ops` property. I think it would be more idiomatic to check the type here.
```suggestion
    if not isinstance(op, Hamiltonian):
```

Reply:
I thought it could be used for `Tensor`, `Prod` and `Sum` or future types. I'll try to be more explicit and add `Sum` support here, since it can also end up in `csr_dot_products` with the current `get_measurement_function` implementation.

Reply:
`Prod` and `Sum` have a `_pauli_rep` property if they are pauli sentences:
```
>>> qml.sum(qml.PauliX(0), qml.PauliY(0))._pauli_rep
1.0 * X(0)
+ 1.0 * Y(0)
>>> qml.prod(qml.S(0), qml.T(0))._pauli_rep is None
True
```

So if a `Prod` or `Sum` instance has a `_pauli_rep` of `None`, we know its not a pauli sentence.

and `Tensor` has an `obs` property instead of an `ops` property. Don't ask me why.